src/types/GBIFRecords.jl

```diff
@@ -30,15 +30,33 @@ struct GBIFRecord
     issues::Vector{Symbol}
     taxonKey::Union{Missing, Integer}
     rank::Union{Missing, AbstractString}
-    taxon::GBIFTaxon
     generic::Union{Missing, AbstractString}
     epithet::Union{Missing, AbstractString}
     vernacular::Union{Missing, AbstractString}
     scientific::Union{Missing, AbstractString}
     observer::Union{Missing, AbstractString}
     license::Union{Missing, AbstractString}
+    taxon::GBIFTaxon
+end
+
+# We add some taxon properties to the GBIFRecord properties.
+# These will flow through to the Tables.jl interface.
+const TAXON_PROPERTIES = (:kingdom, :phylum, :class, :order, :family, :genus, :species)
```

Comment:
Shouldn't this be an `enum`?

Reply:
Sorry I don't understand why. It's just used in `propertynames` and `getproperty` as a tuple of `Symbol` that's known at compile time. Because `getproperty` is modified to include the properties that are in `taxon`, as the Tables.jl interface uses `getpropery`.

I would actually prefer not to have taxon at all and remove this complexity completely, generating it when needed instead. But I wasn't sure of the reasoning behind it. Could `taxon` be generated on the fly from `GBIFRecords`?

Reply:
It might be generated on the fly _but_ we want to have a `taxon` object, and be able to use the species API. I'm not sure I get your point.

Reply:
What I mean is does it have to be an object actually in the memory layout of `GBIFRecord` like it is curently? 

Instead we could do `GBIFTaxon(record)` whenever we need it for the species API, as its essentially the same thing. Then a lot of this code could be deleted.

Reply:
Any interest in this change? it will be a lot cleaner. 

I can write it up fairly easily.